content/ch-applications/hhl_tutorial.ipynb

```diff
@@ -67,7 +71,20 @@
    "source": [
     "## 2. The HHL algorithm<a id='hhlalg'></a>\n",
     "\n",
-    "### A. Some mathematical background<a id='mathbackground'></a>\n",
+    "### A. Conditions <a id='conditions'></a>\n",
+    "The given Linear system has to have a low condition number ${\\displaystyle \\kappa },$ and the Matrix $A$ must be $s$-sparse. This means $A$ must have at most $s$ non-zero entries per row or column. Solving an $s$-sparse system of size $N$ with a classical computer requires $\\mathcal{ O }(Ns\\kappa\\log(1/\\epsilon))$ running time using the conjugate gradient method <sup>[2](#conjgrad)</sup>. Here, $\\epsilon$ deonotes the accuracy of the approximation.\n",
+    " \n",
+    "It is assumed that the user of the algorithm is interested in the result of a scalar measurement on the solution vector, instead of the values of the solution vector itself. So, it assumes that the user is not interested in the values of ${\\displaystyle {\\overrightarrow {x}}}$ itself, but rather the result of applying some operator ${\\displaystyle M}$ onto x, ${\\displaystyle \\langle x|M|x\\rangle }$. Hence, while the classical algorithm returns the full solution, the HHL can only approximate functions of the solution vector.\n",
+    "\n",
+    "Also, Matrix ${\\displaystyle A}$ should be Hermitian so that it can be converted into a unitary operator, under the assumptions of efficient oracles for loading the data, Hamiltonian simulation and computing a function of the solution. \n",
+    "\n",
+    "### B. Significance in Machine Learning<a id='ml'></a>\n",
+    "The time required for a least-squares fit (a continuous function to model a set of discrete points) has been found to become very large as the amount of discrete points increases. However, a research group has developed a quantum algorithm that incorporates HHL to solve many cases of least-squares efficiently, eliminating the need for the previously best higher-complexity algorithm <sup> [3] (#ml) </sup>.\n",
+    "\n",
+    "Of course, HHL is thought to be massively useful in machine learning, where classical algorithms are limited by a polynomial dependence on the volume of data and dimensions of the space. One specific use for HHL in machine learning is in an optimized linear or non-linear binary classifier, or support vector machine. Research has shown that a quantum support vector machine can be used for big data classification and achieve an exponential speedup over classical computers. A quantum algorithm has also been developed for Bayesian training of deep neural networks, with an exponential speedup over classical training due to the use of HHL.\n",
```

Comment:
The following research papers supported the facts in this section. However, they have been summarised and generalised. I can edit this to be more technical, but it may not get the point across to beginners clearly.

https://onlinelibrary.wiley.com/doi/epdf/10.1002/andp.202200082 and https://arxiv.org/pdf/1511.02306.pdf 
https://arxiv.org/pdf/1511.02306.pdf
https://www.researchgate.net/publication/258085144_Quantum_Algorithm_for_Data_Fitting

Reply:
Done.